_episodes_rmd/10-functions.Rmd

```diff
@@ -165,10 +167,110 @@ kelvin_to_celsius <- function(temp) {
 > {: .solution}
 {: .challenge}
 
+## Interlude: Defensive Programming
 
-We're going to define
-a function that calculates the Gross Domestic Product of a nation from the data
-available in our dataset:
+Now that we've begun to appreciate how writing functions provides an efficient
+way to make R code re-usable and modular, we should note that it is important
+to ensure that functions only work in their intended use-cases. This  is a part
+of the concept of _defensive programming_. Defensive programming encourages us
+to frequently check conditions and throw an error if something is
+wrong. These checks are referred to as assertion statements because we want to
+assert some condition is `TRUE` before proceeding. This makes it easier to
+debug because we have a better idea of where the error originated.
+
+### Checking conditions with `stopifnot`
+
+Let’s start by re-examining `fahr_to_kelvin`, our function for converting
```

Comment:
It's been our style to always add the `()` after function names, so it's clear that we are referring to a function.